crates/sui-json/src/lib.rs

```diff
@@ -337,9 +337,7 @@ fn check_valid_homogeneous_rec(curr_q: &mut VecDeque<&JsonValue>) -> Result<(),
     let mut level_type = ValidJsonType::Any;
 
     // Process all in this queue/level
-    while !curr_q.is_empty() {
-        // Okay to unwrap since we know values exist
-        let v = curr_q.pop_front().unwrap();
+    while let Some(v) = curr_q.pop_front() {
```

Comment:
if this is None, should we return `SuiJsonValueError` instead of just exiting the loop? cc @oxade

Reply:
This is more like an invariant violation tbh because we checked above that the queue is non-empty.
We could introduce yet another error type?

But at the core this would be a huge bug, at which point a panic is warranted?